Initial version of i18n in a Magento2 context (#72)
front-commerce/developers.front-commerce.com: source/docs/magento2/i18n.md

---
id: magento2-i18n
title: Magento2 i18n and l10n
---

> Front-Commerce is build with internationalization (<abbr title="internationalization">i18n</abbr>) in mind.

Reviewer: ```suggestion
Front-Commerce is built with internationalization (<abbr title="internationalization">i18n</abbr>) in mind.
```

You can translate every message from your Front-Commerce application as detailed in the [Translate your application](/docs/advanced/theme/translations.html) page.

> If you are running a multi-language store, you can [configure a store per language](/docs/advanced/production-ready/multistore.html).

Reviewer: Would it make sense to prepend this sentence by "In magento" or something like this? This sentence feels a bit odd to me otherwise.

> This sections details what are the steps to take when using Front-Commerce with Magento 2 to build a fully localized store.

Reviewer: ```suggestion
This section details what are the steps to take when using Front-Commerce with Magento 2 to build a fully localized store.
```

## Where should a message be translated?

It is important to understand where a message comes from in order to know where it should be translated.
In a micro-service architecture, information may come from a wide range of data source and Front-Commerce is sometimes only forwarding a message from your remote services.

### Front-Commerce messages

> Terms used in a component (static terms) are translated in Front-Commerce.

Reviewer: ```suggestion
Terms defined in a component are translated in Front-Commerce directly.
```

Please refer to [Translate your application](/docs/advanced/theme/translations.html) to learn more about it.

### Magento2 dynamic content

Dynamic content managed by administrators such as product descriptions, CMS pages, etc. can be found in the Magento admin area.
On a product edition page for instance, you will have to edit the description in the context of the store view used by Front-Commerce (see below).

> You could find more information on Magento documentation: [Translating Products](https://docs.magento.com/m2/ee/user_guide/catalog/product-translate.html) and [Translating Content](https://docs.magento.com/m2/ee/user_guide/cms/page-translate.html).

Reviewer: ```suggestion
You can find more information in Magento's documentation: [Translating Products](https://docs.magento.com/m2/ee/user_guide/catalog/product-translate.html) and [Translating Content](https://docs.magento.com/m2/ee/user_guide/cms/page-translate.html).
```
